tests/integration/sdk/test_vault.py

```
@@ -215,6 +231,50 @@ def test_invalid_credentials_in_local_agent():
 # ============================================
 
 
+def test_session_form_fill_with_vault_without_observe_uses_live_url():
+    """Test form_fill credential replacement snapshots the live page before vault lookup."""
+    _ = load_dotenv()
+    client = NotteClient(api_key=os.getenv("NOTTE_API_KEY"))
+    base_url = "https://apartment-board-demo-nine.vercel.app"
+    login_url = f"{base_url}/auth/signin"
+    username = "alder"
+    password = "rentals123"  # pragma: allowlist secret
```

Ruff (0.15.13)

[error] Possible hardcoded password assigned to: "password"

(S105)